### ISSUE TITLE
Maximum stack size exceeded on >= iojs-3.3.0

### ISSUE THREAD

**Issue by Moeriki** [elastic/elasticsearch-js#265]
Considering this very simple example.

```javascript
var es = require('elasticsearch');
var client = new es.Client({
    apiVersion: '1.5',
    host: process.env.ESHOST,
});

var params = {
    index: 'myIndex',
    type: 'myType',
    body: { data: 'yes' },
};

client.index(params, function (err, result) {
    console.error(err);
    console.log(result);
});
```

I get an error on iojs-v3.3.0 and higher. Tested following releases.

node 0.10.x OK
node 0.12.x OK
iojs-v3.0.0 OK
iojs-v3.3.0 NOT OK
node v4.0 NOT OK

The error in question.

```
[my-project]/node_modules/elasticsearch/src/lib/log.js:65
Log.prototype.listenerCount = function (event) {
                                       ^

RangeError: Maximum call stack size exceeded
    at Log.listenerCount ([my-project]/node_modules/elasticsearch/src/lib/log.js:65:40)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount ([my-project]/node_modules/elasticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount ([my-project]/node_modules/elasticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount ([my-project]/node_modules/elasticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount ([my-project]/node_modules/elasticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount ([my-project]/node_modules/elasticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount ([my-project]/node_modules/elasticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
    at Log.listenerCount ([my-project]/node_modules/elasticsearch/src/lib/log.js:68:25)
    at Function.EventEmitter.listenerCount (events.js:399:20)
```

CC @sintendo

**Comment by Salakar** [comment]
+1 same here on io.js 3.3.0 / node 4.0

**Comment by Ehesp** [comment]
Also having this issue... Same version as above.

**Comment by apechimp** [comment]
This appears to be related to f2a0cede2eff793dd894ee8bef2a2e051f6bafe6

**Comment by spalger** [comment]
@Moeriki @Salakar @Ehesp @apechimp what version of the client are you using? Can you upgrade to the latest version?

**Comment by apechimp** [comment]
Yes. This is fixed in the latest version. As soon as I posted that comment I realized that that commit actually fixes this issue :flushed:. Apologies.

**Comment by spalger** [comment]
np :)

**Comment by Moeriki** [comment]
I was wildly out-of-date as well with 4.0.x — case closed.

**Comment by mik01aj** [comment]
I can confirm this is fixed in 8.2.

**Comment by AseedUsmani** [comment]
I am getting the same on 5.0.0 ???
